### PR TITLE
[tbb] fix tbb by updating it

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Comment by mathisloge** [comment]
related https://github.com/microsoft/vcpkg/pull/19909 and https://github.com/microsoft/vcpkg/pull/23058

File: ports/gazebo/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO osrf/gazebo
-    REF 382dcc3f36095a8d79b5bc9c8b8ad346e867c51d
+    REF ea956014ed45906dda21e1ee682d297189cb4a7b
     SHA512 57638cd0b23b5f2bfd32fdc159d6cd77ca34e3bd695c225591979aef4b7271eac93d3706fa1ffa2340f90013267a4171bebe1e4c142f19ad2bf67963dfed627e
```
Comment (Neumann-A) [inline ports/gazebo/portfile.cmake:5]: Hmm why isn't CI choking on this? Is the hash check broken somehow? cc @BillyONeal @strega-nil

**Comment by JackBoosY** [comment]
See why tbb is not updated in https://github.com/microsoft/vcpkg/pull/16321.

**Comment by Neumann-A** [comment]
@JackBoosY: the only one blocking this is the `usd` port and ras0219 put in on the skip list in #19909

**Comment by Neumann-A** [comment]
Gazebo error:
```
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(58): error C2516: 'tbb::v1::task': is not a legal base class
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(73): error C2143: syntax error: missing ';' before '*'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(73): error C2886: 'tbb::v1::task': symbol cannot be used in a member using-declaration
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(74): error C2334: unexpected token(s) preceding '{'; skipping apparent function body
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(313): error C2039: 'allocate_root': is not a member of 'tbb::v1::task'
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(294): note: This diagnostic occurred in the compiler generated function 'void gazebo::transport::Connection::OnReadData(const boost::system::error_code &,boost::tuples::tuple<Handler,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type>)'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(315): error C2039: 'enqueue': is not a member of 'tbb::v1::task'
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo\transport\Connection.hh(294): note: This diagnostic occurred in the compiler generated function 'void gazebo::transport::Connection::OnReadData(const boost::system::error_code &,boost::tuples::tuple<Handler,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type,boost::tuples::null_type>)'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(40): error C2516: 'tbb::v1::task': is not a legal base class
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(54): error C2143: syntax error: missing ';' before '*'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(54): error C2886: 'tbb::v1::task': symbol cannot be used in a member using-declaration
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(55): error C2334: unexpected token(s) preceding '{'; skipping apparent function body
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(162): error C2039: 'allocate_root': is not a member of 'tbb::v1::task'
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(158): note: This diagnostic occurred in the compiler generated function 'void gazebo::transport::Node::Publish(const std::string &,const google::protobuf::Message &)'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(165): error C2039: 'enqueue': is not a member of 'tbb::v1::task'
D:\installed\x86-windows\include\tbb\../oneapi/tbb/task.h(26): note: see declaration of 'tbb::v1::task'
D:\buildtrees\gazebo\src\7189cb4a7b-e170e5c60c.clean\gazebo/transport/Node.hh(158): note: This diagnostic occurred in the compiler generated function 'void gazebo::transport::Node::Publish(const std::string &,const google::protobuf::Message &)'
```

**Comment by Neumann-A** [comment]
pagmo2 error:

```
FAILED: CMakeFiles/pagmo.dir/src/algorithms/mbh.cpp.obj 
D:\installed\x86-windows\include\boost/mpl/print.hpp(55): error C2220: the following warning is treated as an error
... long template chain
D:\installed\x86-windows\include\boost/mpl/print.hpp(55): warning C4308: negative integral constant converted to unsigned type
```

**Comment by JackBoosY** [comment]
Can I help you to fix these regressions?

**Comment by Neumann-A** [comment]
> Can I help you to fix these regressions?

@JackBoosY: You can try. I won't be able to do anything here until the weekend probably,.

**Comment by Thomas1664** [comment]
> pagmo2 error:
> 
> ```
> FAILED: CMakeFiles/pagmo.dir/src/algorithms/mbh.cpp.obj 
> D:\installed\x86-windows\include\boost/mpl/print.hpp(55): error C2220: the following warning is treated as an error
> ... long template chain
> D:\installed\x86-windows\include\boost/mpl/print.hpp(55): warning C4308: negative integral constant converted to unsigned type
> ```

This can be fixed by patching out `-Werror` or` /WX`

**Comment by Tradias** [comment]
I already wrote the `-Werror` patch in ras0219-msft's pull request: https://github.com/microsoft/vcpkg/pull/19909#discussion_r725655614

**Comment by JackBoosY** [comment]
Any progress here?

**Comment by Neumann-A** [comment]
superseded by #26284

**Pull request closed without merging**